src/main/cpp/src/NativeParquetJni.cpp

```diff
@@ -221,21 +231,36 @@ class column_pruner {
       auto found       = children.find(name);
 
       if (found != children.end()) {
-        // found a match so update the number of children that passed the filter and ask it to
-        // filter itself.
-        ++schema_num_children[our_num_children_index];
+        // Record the current size of schema_map before processing the child
+        // This helps determine if the child was retained after processing.
+        std::size_t before_child_size = schema_map.size();
```

Comment:
Nit:  We can get away with making this `const`.
```suggestion
        auto const before_child_size = schema_map.size();
```

Reply:
Thanks @mythrocks for the review. Updated it. PTAL.